### DISCUSSION TITLE
Helper types for router

### DISCUSSION THREAD

**Discussion by antonio-ivanovski** [TanStack/router#1101] [Ideas]
In the past, I have used `RouteApi<'my-route/path/$someId'>` to inject `routeApi` in some component or external method. This came very handy when there were components that were shared by multiple routes (for example `blog/posts/$postId/edit` and `blog/posts/add`). Currently the `RouteApi` is deprecated so I resort to using `ReturnType<typeof getRouteApi<'....`. Using this is very long and noisy.

Propose that there are types for most of the concepts in the router, as mentioned above for `RouteApi` but also it would be very helpful if we had:
- `RouteLoadingData<'...'>` - get the type of the loader data for the particular path
- `RouteContext<'...'>` - get the type of the context for the particular path
- ... you get the idea

**Comment by BrendanC23** [comment]
I have the same question. Did you ever find a solution? I asked something similar [here](https://github.com/TanStack/router/discussions/1732).
